Tests/kaas/k8s-default-storage-class/helper.py

```diff
@@ -56,19 +56,17 @@ def setup_k8s_client(kubeconfigfile=None):
 
 
 def gen_sonobuoy_result_file(error_n: int, error_msg: str, test_file_name: str):
-
     test_name = test_file_name.replace(".py", "")
-
     test_status = "passed"
-
     if error_n != 0:
         test_status = test_name + "_" + str(error_n)
+        result_file = manual_result_file_template
+        result_file["name"] = test_name
+        result_file["status"] = test_status
+        result_file["details"]["messages"] = error_msg
 
-    result_file = manual_result_file_template
-
-    result_file["name"] = test_name
-    result_file["status"] = test_status
-    result_file["details"]["messages"] = error_msg
+        directory_path = os.path.dirname(f"./{test_name}.result.yaml")
+        os.makedirs(directory_path, exist_ok=True)
 
-    with open(f"./{test_name}.result.yaml", "w") as file:
-        yaml.dump(result_file, file)
+        with open(f"./{test_name}.result.yaml", "w") as file:
+            yaml.dump(result_file, file)
```

Comment:
This still creates a file hierarchy replicating the absolute path to the script file, e.g. for me it creates:

`/home/martinmorgenstern/workspace/scs/standards/Tests/home/martinmorgenstern/workspace/scs/standards/Tests/kaas/k8s-default-storage-class`

The reason is that `__file__` (which is the absolute path to the script) is used as the `test_file_name` argument passed to `gen_sonobuoy_result_file` and used without further processing.

In `main()`, you should use `os.path.basename(__file__)` to get only the script filename without the directories.

Reply:
yes you're right. ithought i already changed that